ergotree-ir/src/sigma_protocol/sigma_boolean/cand.rs

```diff
@@ -45,11 +48,55 @@ impl Cand {
     }
 }
 
+impl SigmaSerializable for Cand {
+    fn sigma_serialize<W: SigmaByteWrite>(&self, w: &mut W) -> SigmaSerializeResult {
+        self.items.sigma_serialize(w)
+    }
+
+    fn sigma_parse<R: SigmaByteRead>(r: &mut R) -> Result<Self, SigmaParsingError> {
+        let items = SigmaConjectureItems::<_>::sigma_parse(r)?;
+        Ok(Cand { items })
+    }
+}
+
+impl From<Vec<SigmaBoolean>> for Cand {
```

Comment:
We should avoid using unwrap in production code. It's perfectly fine in tests though. 
`From` is not applicable here, since we want the error on an invalid number of items to be handled by the caller. `TryFrom` fits here, but be aware that `BoundedVec` (type behind a `SigmaConjectureItems` type alias) has one.

Reply:
Makes sense.  So I am removing this code from `Cand`/`Cor`, and I am okay to unwrap() in the `SigmaBoolean` Arbitrary impl as below (since the caller is test code and does not need to handle the error gracefully). 
```
Cand { 
   items: BoundedVec::try_from(items).unwrap() 
}
```

Will check in both changes.